### PR TITLE
Introducing: The `kalite` command, much awesome

### PR DESCRIPTION
**Pull request by benjaoming**
Try it out!

    ➜  ka-lite git:(kalite-command) bin/kalite --help
    KA Lite (Khan Academy Lite)

    Supported by Foundation for Learning Equality
    www.learningequality.org

    Usage:
      kalite start [options] [--skip-job-scheduler] [DJANGO_OPTIONS ...]
      kalite stop [options] [DJANGO_OPTIONS ...]
      kalite status [job-scheduler] [options]
      kalite shell [options] [DJANGO_OPTIONS ...]
      kalite test [options] [DJANGO_OPTIONS ...]
      kalite manage COMMAND [options] [DJANGO_OPTIONS ...]
      kalite -h | --help
      kalite --version

    Options:
      -h --help             Show this screen.
      --version             Show version.
      COMMAND               The name of any available django manage command. For
                            help, type `kalite manage help`
      --debug               Output debug messages (for development)
      --skip-job-scheduler  For `kalite start`: Skips running the job scheduler
                            (useful for dev)
      DJANGO_OPTIONS        All options are passed on to the django manage command.
                            Notice that all django options must be place *last* and
                            should not be mixed with other options.

Try for instance to start kalite, and while its starting, hurry up and fire off the status command.

# TODO

- [ ] Testing on Win
- [x] Testing on OSX
- [ ] Testing on Android
- [x] Testing on Linux (Ubuntu)
- [x] Test case for .travis
- [x] Bat script for Windows, currently do: `python.exe bin/kalite`
- [ ] Resolve how implementation should work with service / task / rc.d etc systems.
- [x] Document SHELL environment variables
- [ ] Add a `kalite setup` command

### CODE REVIEW

**Comment by aronasorman** [comment]
Cool, tests pass :)

File: bin/kalite [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+"""
+Role of this executable:
+Determine where kalite is installed and which python to use. This can be
+set in the system's env (useful for distribution).
+
+All arguments are proxied for kalite, and all stderr and stdout is returned.
+
+Notes...
+THIS MAY BE RUN BY PYTHON 2 AND 3 !!!
```
Comment (aronasorman) [inline bin/kalite]: Nice!

File: kalite/distributed/management/commands/kaserve.py [hunk 2]
```diff
@@ -57,6 +57,13 @@ class Command(BaseCommand):
             help="PID file"
         ),
         make_option(
+            '--startup-lock-file',
```
Comment (aronasorman) [inline kalite/distributed/management/commands/kaserve.py]: Can you put in more documentation on how this startup lock file is used?

**Comment by aronasorman** [comment]
@benjaoming can document the different environment variables that can affect the `kalite` executable (such as `KALITE_DIR`) in the command's help text?

**Comment by aronasorman** [comment]
should be useful for @cpauya

**Comment by aronasorman** [comment]
Did a first passthrough of this. Will take a look again when my brain isn't in pieces.

**Comment by benjaoming** [comment]
I made a brain fart regarding Windows execution... of course Windows doesn't read shebang. I'm coming up with a bat alternative shortly... too bad it had to end this way, was almost certain that we would get rid of operating system specific code.

I'm contemplating, though, that we can replace a `kalite.bat` with some text in the installation instructions, asking Windows users to double-click a 'setup.bat' file which will create a `doskey` command called 'kalite' as in this example -> http://stackoverflow.com/a/21040825/405682

File: bin/kalite [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import sys
+import subprocess
+from distutils import spawn
+
+kalite_path = None
+
+# Yay! The path is set!
+if 'KALITE_DIR' in os.environ:
+    kalite_path = os.environ['KALITE_DIR']
+    if os.path.exists(kalite_path):
```
Comment (cpauya) [inline bin/kalite]: I think we should use `if not os.path.exists` here?

Comment (benjaoming) [inline bin/kalite]: Oh jeez, yes you're right! Fixed in 63cf6c7

**Comment by aronasorman** [comment]
Tested on the mac. `bin/kalite manage setup`, `bin/kalite start` and `bin/kalite stop` works as expected.

For the `setup` command, it might be good to remove this output:
```
2611 static files copied.
Scanning for video files in the content directory (/Users/aron/src/github.com/aronasorman/ka-lite/content/)

CONGRATULATIONS! You've finished setting up the KA Lite server software.
	Please run '/Users/aron/src/github.com/aronasorman/ka-lite/start.command' to start the server,
	then load one of the following addresses in your browser to complete the configuration:
		http://192.168.2.1:8008/
		http://192.168.2.138:8008/
		http://127.0.0.1:8008/
		http://192.168.53.1:8008/
```

since the server isn't running yet at that time.

**Comment by aronasorman** [comment]
I get this error when running in python 3:
```
KALITE_PYTHON=`which python3` bin/kalite start
Traceback (most recent call last):
  File "/Users/aron/src/github.com/aronasorman/ka-lite/kalitectl.py", line 36, in <module>
    from django.core.management import execute_from_command_line
  File "python-packages/django/core/management/__init__.py", line 5, in <module>
    import imp
  File "/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/imp.py", line 19, in <module>
    from importlib._bootstrap import SourcelessFileLoader, _ERR_MSG, _SpecMethods
ImportError: No module named 'importlib._bootstrap'
```

**Comment by aronasorman** [comment]
Tried the bare command, got this output:
```
$ bin/kalite
Usage:
  kalite start [options] [--skip-job-scheduler] [DJANGO_OPTIONS ...]
  kalite stop [options] [DJANGO_OPTIONS ...]
  kalite status [job-scheduler] [options]
  kalite shell [options] [DJANGO_OPTIONS ...]
  kalite test [options] [DJANGO_OPTIONS ...]
  kalite manage COMMAND [options] [DJANGO_OPTIONS ...]
  kalite -h | --help
  kalite --version
``

**Comment by benjaoming** [comment]
@MCGallaspy Can you try out things on your virtualbox's host OS? :)

**Comment by cpauya** [comment]
Hi @benjaoming, I tried your latest commits and got the following error:

```bash
(ka-lite)cyril-lappy6:ka-lite cyril$ kalite status
Traceback (most recent call last):
  File "/Users/cyril/w/fle/benjaoming/ka-lite/kalitectl.py", line 397, in <module>
    sys.stderr.write("{0:s} ({1:d})\n".format(status_code, verbose_status))
ValueError: Unknown format code 's' for object of type 'int'
(ka-lite)cyril-lappy6:ka-lite cyril$ sudo cp bin/kalite /usr/bin/kalite
Password:
(ka-lite)cyril-lappy6:ka-lite cyril$ kalite status
Traceback (most recent call last):
  File "/Users/cyril/w/fle/benjaoming/ka-lite/kalitectl.py", line 397, in <module>
    sys.stderr.write("{0:s} ({1:d})\n".format(status_code, verbose_status))
ValueError: Unknown format code 's' for object of type 'int'
(ka-lite)cyril-lappy6:ka-lite cyril$ p
/Users/cyril/w/fle/benjaoming/ka-lite
(ka-lite)cyril-lappy6:ka-lite cyril$
```

As of earlier, my issue was that the the `kalite` command can only be used on the directory of the `kalitectl.py` script.  Here's a series of commands I did on my terminal for your reference (I sudo copied `bin/kalite` to `/usr/bin/kalite`):

```bash
(ka-lite)cyril-lappy6:ka-lite cyril$ kalite status
1 Stopped
(ka-lite)cyril-lappy6:ka-lite cyril$ kalite manage testx
Unknown command: 'testx'
Type 'kalitectl.py help' for usage.
(ka-lite)cyril-lappy6:ka-lite cyril$ ..
(ka-lite)cyril-lappy6:benjaoming cyril$ p
/Users/cyril/w/fle/benjaoming
(ka-lite)cyril-lappy6:benjaoming cyril$ kalite manage testx
Traceback (most recent call last):
  File "/Users/cyril/w/fle/benjaoming/ka-lite/kalitectl.py", line 36, in <module>
    from django.core.management import execute_from_command_line
ImportError: No module named django.core.management
(ka-lite)cyril-lappy6:benjaoming cyril$ kalite status
Traceback (most recent call last):
  File "/Users/cyril/w/fle/benjaoming/ka-lite/kalitectl.py", line 36, in <module>
    from django.core.management import execute_from_command_line
ImportError: No module named django.core.management
```

For all those, I already have the `KALITE_DIR` environment variable set as:

```bash
(ka-lite)cyril-lappy6:ka-lite cyril$ env | grep KA
KALITE_DIR=/Users/cyril/w/fle/benjaoming/ka-lite/
(ka-lite)cyril-lappy6:ka-lite cyril$
```

Is there something am missing?

**Comment by benjaoming** [comment]
@cpauya 
Thanks for testing again!! I fixed the err in status, was introduced by an idiotic string formatting using positional arguments, bad habit.

I am not sure what the `kalite testx` command is? Typo? :)

**Comment by benjaoming** [comment]
Your `KALITE_DIR` looks correct and things seem to work on your system apart from the above few things..

**Comment by cpauya** [comment]
Nope, I did that on purpose so I won't have a lot results from running the `kalite manage <command>` :smile_cat: 

Yes, when it's run on the same directory, it works.

I had to copy `bin/kalite` though, tried symlinking yesterday with `sudo ln -s bin/kalite /usr/bin/kalite but it didn't work.

**Comment by cpauya** [comment]
Yes, it's working now!  Thanks for the quick fix. :+1:

**Comment by benjaoming** [comment]
@cpauya Thanks for quick feedback!! Please do stress test the new start/stop/status commands, if you have time...

**Comment by aronasorman** [comment]
I believe we've made a travis "test case", so I marked it as checked in your todo list @benjaoming

**Comment by benjaoming** [comment]
@aron, agreed, the basic test of start/status/stop should be sufficient for everyday testing.

@EdDixon @ruimalheiro 
I need some help with the Windows test cases. Yesterday, I had 'kalite' on my path in wine, and I could execute from any working directory on the command line.

But every time I close my cmd, I have to run `bin/windows/kalitesetup.bat` again.

Can you test, that if you run `kalitesetup.bat`, that the `kalite` command also works after restarting the command line / restarting the machine?

**Comment by cpauya** [comment]
Hi @benjaoming - your most recent updates broke the `kalite status` with this:

```bash
(ka-lite)cyril-lappy6:ka-lite cyril$ kalite status
Traceback (most recent call last):
  File "/Users/cyril/w/fle/benjaoming/ka-lite/kalitectl.py", line 422, in <module>
    if arguments['job-scheduler']:
KeyError: 'job-scheduler'
(ka-lite)cyril-lappy6:ka-lite cyril$
```

**Comment by benjaoming** [comment]
Grrr, you're right... and actually it reveals a problem in the travis test...

    $ bin/kalite status
    Traceback (most recent call last):
    File "/home/travis/build/learningequality/ka-lite/kalitectl.py", line 422, in <module>
    if arguments['job-scheduler']:
    KeyError: 'job-scheduler'
    
    The command "bin/kalite status" exited with 0.

**Comment by benjaoming** [comment]
So now the travis test should work again since `kalite` now returns the status of `kalitectl`.

**Comment by aronasorman** [comment]
Windows bugs notwithstanding, what are the other blockers for this?

**Comment by aronasorman** [comment]
Merging this in, so everyone can test it :dancers:

**Comment by EdDixon** [comment]
I plan to work on this over the weekend, made some headway with the windows scripts.

**Pull request merged**